CMakeLists.txt

```diff
@@ -562,9 +565,15 @@ if (BUILD_TESTING)
 
     file(GLOB UNITTESTS_SRC "tests/unit/*.c")
     foreach(test_case ${UNITTESTS_SRC})
+    foreach(fips_mode ${TEST_FIPS_MODES})
         # NAME_WE: name without extension
         get_filename_component(test_case_name ${test_case} NAME_WE)
 
+        if(fips_mode)
+            set(test_case_name ${test_case_name}+fips)
+            set(target_compile_definitions(${test_case_name} PRIVATE S2N_TEST_IN_FIPS_MODE))
```

Comment:
I can't figure out why this is wrapped in a `set` directive. Can we add a comment?

Reply:
Looking at the code, it looks like there is an `ifdef` in `s2n_test.h`, so I think this compile definition also needs to be added to the testlib.

Maybe it would be neater as a `PUBLIC` option on the testlib? But perhaps that's a bit too magical.